zokrates_ast/src/ir/serialize.rs

```diff
@@ -1,14 +1,19 @@
-use crate::ir::check::UnconstrainedVariableDetector;
+use crate::{
+    ir::{check::UnconstrainedVariableDetector, solver_indexer::SolverIndexer},
+    Solver,
+};
 
 use super::{ProgIterator, Statement};
+use byteorder::{LittleEndian, ReadBytesExt, WriteBytesExt};
+use serde::Deserialize;
 use serde_cbor::{self, StreamDeserializer};
-use std::io::{Read, Write};
+use std::io::{Read, Seek, Write};
 use zokrates_field::*;
 
 type DynamicError = Box<dyn std::error::Error>;
 
 const ZOKRATES_MAGIC: &[u8; 4] = &[0x5a, 0x4f, 0x4b, 0];
-const ZOKRATES_VERSION_2: &[u8; 4] = &[0, 0, 0, 2];
+const FILE_VERSION: &[u8; 4] = &[3, 0, 0, 0];
```

Comment:
I feel this doesn't really represent a zokrates version, it is more of a file version. I changed it to a little-endian order to align with everything else.